src/EventLoop/Internal/AbstractDriver.php

```diff
@@ -615,10 +619,19 @@ protected function invokeCallback(Callback $callback): void
     protected function error(\Throwable $exception): void
     {
         if ($this->errorHandler === null) {
-            throw $exception;
+            $this->interrupt = static fn () => throw $exception;
+            return;
         }
 
-        ($this->errorHandler)($exception);
+        $fiber = new \Fiber(function (callable $errorHandler, \Throwable $exception): void {
+            try {
+                $errorHandler($exception);
+            } catch (\Throwable $exception) {
+                $this->interrupt = static fn () => throw $exception;
```

Comment:
I think we should keep the method we had and add an assertion that `$this->interrupt` is `null` before setting it.

Reply:
Ok, yeah, `$this->interrupt` should always be null.